platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Popup.kt

```diff
@@ -18,7 +21,102 @@ import androidx.compose.ui.unit.IntOffset
 import androidx.compose.ui.unit.IntRect
 import androidx.compose.ui.unit.IntSize
 import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.window.Popup as ComposePopup
 import androidx.compose.ui.window.PopupPositionProvider
+import androidx.compose.ui.window.PopupProperties
+import org.jetbrains.jewel.foundation.InternalJewelApi
+import org.jetbrains.jewel.foundation.JewelFlags
+import org.jetbrains.jewel.ui.component.PopupRenderer.PopupData
+
+/**
+ * Displays a popup with the provided content at a position determined by the given [PopupPositionProvider].
+ *
+ * This function behavior is influenced by the 'jewel.customPopupRender' system property. If set to `true`, it allows
+ * using a custom popup rendering implementation; otherwise, it defaults to the standard Compose popup.
+ *
+ * If running on the IntelliJ Platform and setting the [JewelFlags.useCustomPopupRenderer] property to `true`, the
+ * plugin will use the JBPopup implementation for rendering popups. This is useful if your composable content is small,
+ * but you need to display a popup that is bigger than the component size.
+ *
+ * @param popupPositionProvider Determines the position of the popup on the screen.
+ * @param onDismissRequest Callback invoked when a dismiss event is requested, typically when the popup is dismissed.
+ * @param properties Configuration parameters for the popup, such as whether it should consume touch events or focusable
+ *   behavior.
+ * @param onPreviewKeyEvent Callback invoked for key events before they are dispatched to children. Return `true` to
+ *   consume the event.
+ * @param onKeyEvent Callback invoked for key events after they are dispatched to children. Return `true` to consume the
+ *   event.
+ * @param content The composable content to be displayed inside the popup.
+ */
+@Composable
+@InternalJewelApi
```

Comment:
I don't think this should be internal, but rather, it should be experimental

Reply:
I don't think so. It's an implementation detail. If they use Popup and the flag is on they get the behaviour which is the important part. If we find out a use case for it to be opened up we can do it later on.